lib/ansible/modules/windows/win_zip.ps1

```diff
@@ -0,0 +1,253 @@
+            }
+            catch {
+                Remove-BrokenZip $dest
+                Fail-Json -obj $result -message "Error creating zip file '$dest': $($_.Exception.Message)"
+            }
+        }
+        "DotNet" {
+            try {
+                # add type here is only needed for force testing dotnet in the calling main code
+                Add-Type -AssemblyName System.IO.Compression.FileSystem -ErrorAction Stop| Out-Null
+                Add-Type -AssemblyName System.IO.Compression -ErrorAction Stop | Out-Null
+
+                if (Test-Path -LiteralPath $src -PathType Container)
+                {
+                    if (-not $check_mode) {
+                        [System.IO.Compression.ZipFile]::CreateFromDirectory($src, $dest)
+                    }
+                    #$result.changed = $true
+                }
+                else {
+                    if (-not $check_mode) {
+                        $archive = [System.IO.Compression.ZipFile]::Open($dest, [System.IO.Compression.ZipArchiveMode]::Create, [System.Text.Encoding]::UTF8)
+                        $filename = (Get-Item -LiteralPath $src).Name
+                        [System.IO.Compression.ZipFileExtensions]::CreateEntryFromFile($archive, $src, $filename, 0) | Out-Null
+                    }
+                    #$result.changed = $true
+                }
+            }
+            catch {
+                # If zip is opened but something fails afterwards, it will be left on disk. Need to cleanup
+                # Duplicated here and not just finally because we have to dispose it before we can delete it
+                # or the unmanaged file is locked
+                if ((Get-Variable -Name archive -ErrorAction SilentlyContinue) -and ($null -ne $archive)) {
+                    $archive.Dispose()
+                }
+                Remove-BrokenZip $dest
+                Fail-Json -obj $result -message "Error creating zip file '$dest': $($_.Exception.Message)"
+
+            }
+            finally {
+                # Call dispose to get rid of the unmanaged object
+                if ((Get-Variable -Name archive -ErrorAction SilentlyContinue) -and ($null -ne $archive)) {
+                    $archive.Dispose()
+                }
+            }
+        }
+        "Undefined" { # This should never be hit as we check, but leave in just in case the early check is modified.
+            #Fail-Json -obj $result -message "Could not find any available compression utilities: $($_.Exception.Message)"
+            Fail-Json -obj $result -message "Could not find any available compression utilities: PowerShell 5.0 and .NET Framework 4.5 are minimally required.  Please install the Windows Management Framework 5.0 or higher."
+        }
+    }
+
+}
+
+# BEGIN: Determine which type of compression utils to use: PowerShell, .NET or COM (to follow dev standards)
+# Try PowerShell native first, then .NET 4.5
+Add-Type -ErrorAction SilentlyContinue -TypeDefinition @"
```

Comment:
I don't think this is necessary, just use .NET as it's the lowest common denominator in what we support today.